src/PowerManifold.jl

```diff
@@ -0,0 +1,364 @@
+end
+
+
+struct PowerMetric <: Metric end
+
+function det_local_metric(M::MetricManifold{PowerManifold, PowerMetric}, x::AbstractArray)
+    result = one(eltype(x))
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        result *= det_local_metric(M.manifold,
+            _read(rep_size, x, i))
+    end
+    return result
+end
+
+function inner(M::PowerManifold, x, v, w)
+    result = zero(eltype(v))
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        result += inner(M.manifold,
+            _read(rep_size, x, i),
+            _read(rep_size, v, i),
+            _read(rep_size, w, i))
+    end
+    return result
+end
+
+function norm(M::PowerManifold, x, v)
+    sum_squares = zero(eltype(v))
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        sum_squares += norm(M.manifold,
+            _read(rep_size, x, i),
+            _read(rep_size, v, i))^2
+    end
+    return sqrt(sum_squares)
+end
+
+function distance(M::PowerManifold, x, y)
+    sum_squares = zero(eltype(x))
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        sum_squares += distance(M.manifold,
+            _read(rep_size, x, i),
+            _read(rep_size, y, i))^2
+    end
+    return sqrt(sum_squares)
+end
+
+function exp!(M::PowerManifold, y, x, v)
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        exp!(M.manifold,
+            _write(rep_size, y, i),
+            _read(rep_size, x, i),
+            _read(rep_size, v, i))
+    end
+    return y
+end
+
+function log!(M::PowerManifold, v, x, y)
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        log!(M.manifold,
+            _write(rep_size, v, i),
+            _read(rep_size, x, i),
+            _read(rep_size, y, i))
+    end
+    return v
+end
+
+function injectivity_radius(M::PowerManifold, x)
+    radius = 0.0
+    initialized = false
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        cur_rad = injectivity_radius(M.manifold,
+            _read(rep_size, x, i))
+        if initialized
+            radius = min(cur_rad, radius)
+        else
+            radius = cur_rad
+            initialized = true
+        end
+    end
+    return radius
+end
+
+function injectivity_radius(M::PowerManifold)
+    radius = 0.0
+    initialized = false
+    for i in get_iterator(M)
+        cur_rad = injectivity_radius(M.manifold)
+        if initialized
+            radius = min(cur_rad, radius)
+        else
+            radius = cur_rad
+            initialized = true
+        end
+    end
+    return radius
+end
+
+"""
+    PowerRetraction(retraction::AbstractRetractionMethod)
+
+Power retraction based on `retraction`. Works on [`PowerManifold`](@ref).
+"""
+struct PowerRetraction{TR<:AbstractRetractionMethod} <: AbstractRetractionMethod
```

Comment:
Does this type just “power up” a retraction or can that also be a retraction that's different (though I can not imagine a case where I would use a different retraction for components 1 and 2)...

Reply:
This is just the "power up" of a single retraction. If someone needs a different retraction, making a new one is fairly simple.

Reply:
Ok, great.